source/main.c

```diff
@@ -459,27 +466,45 @@ int do_nand_backup()
 
 			// int barwidth = conX - (5 /* ___% */ + 2 /* [] */ + 1 /* to avoid new line lol */);
 			int barwidth = conX - ( 16 /* [ XXXX / XXXX ] */ + 1);
-			float prog = (i + 1) / (float)n_blocks;
+			float prog = i / (float)n_blocks;
 			int r = prog * barwidth;
 
 			// printf("% 3i%% [%.*s%*s]\r", (int)(prog * 100), r, thestring, barwidth - r, "");
-			printf("[ %4u / %4u ] \x1b[42;1m%*s\x1b[40m%*s\r", i + 1, n_blocks, r, "", barwidth - r, "");
+			printf("\x1b[26;0H");
+			printf("[ %4u / %4u ] \x1b[42;30m%*s\x1b[40m%*s\r", i + 1, n_blocks, r, "", barwidth - r, "");
+			printf("\x1b[26;46H");
+			if (i > 2193) {
+				printf("\x1b[42;30m%d", (100*i)/(n_blocks - 1));
+				putchar('%');
```

Comment:
It's possible to print a literal % with printf, see the commented out printf above